gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/CategoryServiceImpl.java

```diff
@@ -218,7 +220,9 @@ public List<CategoryEntity> update(ExecutionContext executionContext, final List
         final List<CategoryEntity> savedCategories = new ArrayList<>(categoriesEntities.size());
         categoriesEntities.forEach(categoryEntity -> {
             try {
-                Optional<Category> optCategoryToUpdate = categoryRepository.findById(categoryEntity.getId());
+                Optional<Category> optCategoryToUpdate = categoryRepository
+                    .findById(categoryEntity.getId())
+                    .filter(category -> category.getEnvironmentId().equalsIgnoreCase(executionContext.getEnvironmentId()));
```

Comment:
orElseGet ?

Reply:
No because we do nothing if we don't find the category.